projects/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/metadata/AnnotationMetadata.kt

```diff
@@ -38,14 +37,15 @@ val SINGLETON = DefinitionAnnotation("single", annotationType = Singleton::class
 val FACTORY = DefinitionAnnotation("factory", annotationType = Factory::class)
 val SCOPE = DefinitionAnnotation("scoped", annotationType = Scope::class)
 val SCOPED = DefinitionAnnotation("scoped", annotationType = Scoped::class)
-val KOIN_VIEWMODEL = DefinitionAnnotation("viewModel", "org.koin.androidx.viewmodel.dsl.viewModel", KoinViewModel::class)
+val KOIN_VIEWMODEL = DefinitionAnnotation("viewModel", "org.koin.androidx.viewmodel.dsl.viewModel", org.koin.android.annotation.KoinViewModel::class)
+val KOIN_VIEWMODEL_KMP = DefinitionAnnotation("viewModel", "org.koin.compose.viewmodel.dsl.viewModel", KoinViewModel::class)
```

Comment:
at the end we call it on the same API, it's just the hosting project that needs to setup the right dependencies 🤔

Reply:
we can't guarantee app always use lifecycle-viewmodel (2.0.0+) with koin-annotation 1.4.0(next release).
therefore we should consider the case use new koin-annotation version with old lifecycle-viewmodel (under 2.0.0).
It means koin-annotation must support two kind of viewmodel (androidMain and commonMain).
If koin-annotation 1.4.0 only support with latest lifecycle-viewmodel (2.0.0+), I think we must upgrade major version rather than minor version because it didn't guarantee backward compatibility.

So I added two versions of the viewmodel definition and delegated to the host to use the correct viewmodel.

The host will use coreMain's KoinViewModel if it uses the latest lifecycle-viewmodel, or androidMain's KoinViewModel if it is not up to date.